From 0f4f5bd25aaae22a2d01a2507b082439a3d9577e Mon Sep 17 00:00:00 2001
From: santoshkumar <santoshkumar.laxminarayan.rai@intel.com>
Date: Thu, 7 Nov 2019 14:43:45 +0800
Subject: [PATCH] Add stateless handling.

Checking sos.conf file path , if sos.conf not present in /etc by default
then add support to pull off Clear Linux configuration files from their stateless
locations.

Taken referance of previous patch created by William and found function
where to make changes,since v3.5 to v3.8 internal functions calling are being
changed.

Author: William Douglas <william.douglas@intel.com>
Re-creator: santoshkumar <santoshkumar.laxminarayan.rai@intel.com>
---
 sos/__init__.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sos/__init__.py b/sos/__init__.py
index 16cb35b4..9288535b 100644
--- a/sos/__init__.py
+++ b/sos/__init__.py
@@ -17,6 +17,7 @@ gettext to internationalize messages.
 
 import gettext
 import six
+import os
 
 from argparse import ArgumentParser
 
@@ -236,6 +237,12 @@ class SoSOptions(object):
     def from_file(cls, argparser, config_file, is_default=True):
         opts = SoSOptions()
         config = ConfigParser()
+
+        if os.path.isfile('/etc/sos.conf'):
+            config_file = '/etc/sos.conf'
+        else:
+            config_file = '/usr/share/defaults/sos/sos.conf'
+
         try:
             try:
                 with open(config_file) as f:
-- 
2.23.0

